rest-client/src/main/java/org/sagebionetworks/bridge/rest/ApiClientProvider.java

```diff
@@ -74,54 +180,49 @@ public UserSessionInfoProvider getUserSessionInfoProvider(SignIn signIn) {
         checkNotNull(service);
         checkNotNull(signIn);
 
+        if (Strings.isNullOrEmpty(signIn.getPassword())) {
+            return getClient(service, signIn.getEmail());
+        }
+
         SignIn internalSignIn = RestUtils.makeInternalCopy(signIn);
         
         // Do not allow more than one session provider to be created for a signIn.
         UserSessionInfoProvider sessionProvider = null;
         synchronized(this) {
-            sessionProvider = sessionProviders.get(internalSignIn);
+            sessionProvider = signInSessionProviders.get(internalSignIn);
             if (sessionProvider == null) {
                 sessionProvider = new UserSessionInfoProvider(getClient(AuthenticationApi.class), internalSignIn);
-                sessionProviders.put(internalSignIn, sessionProvider);
+                emailSessionProviders.put(internalSignIn.getEmail(), sessionProvider);
```

Comment:
Note here that an email based SessionProvider is replaced by the sign-in based one. Prefer sign-in because it's "better" in that it can do the same token reauth as email based, plus fall back to sign-in based.